### PR TITLE
BAH-2998 | Add. Functionality to capture test result 

### PR DESCRIPTION
**Pull request by abinaya-u**
Jira link: https://bahmni.atlassian.net/browse/BAH-2998?atlOrigin=eyJpIjoiMjIyYWNhMzlkM2Y1NGRlMzkzYzFkOGVkNzM5M2ZhZjkiLCJwIjoiaiJ9

Summary:
1. This PR has changes to save the test result value which has the data type as 'Numeric'
2. This PR doesn't cover capturing test result for Panel
3. This Save will work only if attachment is not present. Both attachment and value will not be saved.
4. This PR doesn't take care of Abnormal result.

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportService.java [hunk 1]
```diff
@@ -50,23 +57,20 @@
...
 import java.time.temporal.ChronoUnit;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Date;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Optional;
-import java.util.Set;
-import java.util.UUID;
+import java.util.*;
```
Comment (angshu) [inline api/src/main/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportService.java:68]: replace with single line imports

File: api/src/main/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportService.java [hunk 2]
```diff
@@ -124,43 +128,97 @@ public class ObsBasedDiagnosticReportService extends BaseFhirService<DiagnosticR
...
-			Set<Obs> reportObs = createReportObs(fhirDiagnosticReport, order, encounter);
+			if(attachmentObs.isEmpty()) {
+				Obs obs = reportResults.get(0);
+				LabResult labResult = LabResult.builder().
+						labResultValue(obs.getValueNumeric().toString())
+						.concept(fhirDiagnosticReport.getCode())
+						.obsFactory(newObs(fhirDiagnosticReport.getSubject(), fhirDiagnosticReport.getIssued()))
+						.build();
+				fhirDiagnosticReport.setResults(
+						of(diagnosticReportObsLabResultTranslator.toOpenmrsType(labResult))
```
Comment (angshu) [inline api/src/main/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportService.java:167]: better to write Stream.of(..) .. clearer

File: api/src/test/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportServiceTest.java [hunk 3]
```diff
@@ -5,10 +5,7 @@
 import org.hl7.fhir.r4.model.*;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.mockito.ArgumentCaptor;
-import org.mockito.InjectMocks;
-import org.mockito.Mock;
-import org.mockito.Mockito;
+import org.mockito.*;
```
Comment (angshu) [inline api/src/test/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportServiceTest.java:8]: replace with single line imports

File: api/src/test/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportServiceTest.java [hunk 4]
```diff
@@ -5,10 +5,7 @@
...
 import org.mockito.runners.MockitoJUnitRunner;
 import org.openmrs.*;
```
Comment (angshu) [inline api/src/test/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportServiceTest.java:10]: replace with single line imports

File: api/src/main/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportService.java (hunk 1 above)
Comment (abinaya-u) [reply api/src/main/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportService.java]: changes have been made

File: api/src/main/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportService.java (hunk 2 above)
Comment (abinaya-u) [reply api/src/main/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportService.java]: changes have been made

File: api/src/test/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportServiceTest.java (hunk 3 above)
Comment (abinaya-u) [reply api/src/test/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportServiceTest.java]: changes have been made

File: api/src/test/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportServiceTest.java (hunk 4 above)
Comment (abinaya-u) [reply api/src/test/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportServiceTest.java]: changes have been made

**Review by angshu — APPROVED** [review]

**Pull request merged**